version: "3.8"

x-drf: &drf
  build:
    context: .
    dockerfile: ./docker/geo_backend/Dockerfile
  image: backend
  restart: always
  volumes:
    - ./geo_back/householder_gis/:/app/householder_gis/
    - static-volume:/app/static/
    - ./.env:/app/.env


services:
  drf:
    <<: *drf
    container_name: geo-backend
    command: gunicorn --bind 0.0.0.0:8000 --workers 2 householder_gis.wsgi:application --reload
    expose:
      - "8000"
    ports:
      - "8090:8000"
    extra_hosts:
      - "host.docker.internal:172.17.0.1"
    networks:
      - geo-network


  frontend:
    build:
      context: .
      dockerfile: ./docker/geo_frontend/Dockerfile
    image: frontend
    restart: always
    ports:
      - 3000:3000
    command: npm start
    environment:
      - CHOKIDAR_USEPOLLING=true
    volumes:
      - ./geo_front/src/:/app/src/
      - ./geo_front/node_modules/:/app/node_modules/
      - ./geo_front/build/:/app/build/
      - ./geo_front/public/:/app/public/
      - ./geo_front/package.json:/app/package.json
    networks:
      - geo-network
    depends_on:
      - drf

  drf-test:
    <<: *drf
    container_name: geo-backend-test
    profiles: ["test"]
    command: "pytest --no-cov"
    extra_hosts:
      - "host.docker.internal:172.17.0.1"
    networks:
      - geo-network



volumes:
  static-volume:

networks:
  geo-network:
    name: geo-network
    driver: bridge
